flake.lock

```diff
@@ -0,0 +1,59 @@
+{
+  "nodes": {
+    "flake-utils": {
+      "locked": {
+        "lastModified": 1644229661,
+        "narHash": "sha256-1YdnJAsNy69bpcjuoKdOYQX0YxZBiCYZo4Twxerqv7k=",
+        "owner": "numtide",
+        "repo": "flake-utils",
+        "rev": "3cecb5b042f7f209c56ffd8371b2711a290ec797",
+        "type": "github"
+      },
+      "original": {
+        "owner": "numtide",
+        "repo": "flake-utils",
+        "type": "github"
+      }
+    },
+    "llvm-hs-src": {
```

Comment:
Why do we have this lockfile? Can the definitions in nixpgs change? How do we update it if we do want to use some newer versions?

Reply:
The lockfile is for reproducible builds using Nix flakes. `default.nix` uses the current (global) version of `nixpkgs` on a given machine, which may differ from machine to machine.

On the other hand, a Nix flake will pin all the dependencies, including Nixpkgs. Upstream Nixpkgs may change -- to update dependencies in `flake.lock`, we can run [`nix flake lock`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake-lock.html). In theory, we'd have CI for the Nix flake so if we update it and it breaks, we'll be alerted by CI.